### PR TITLE
Scale PIO test suite

### LINKED ISSUE #181: Scaling the PIO2 test suite
Allow users to specify the number of MPI processes for the entire PIO2 test suite.

All Fortran tests in the PIO2 test suite allow running with any number of MPI processes. Some C unit tests only allow specific number of MPI processes.

This change should allow users to specify (via environment) the number of MPI processes to run the entire test suite.
* First phase : Allow users to run existing scalable tests in the suite by specifying the number of MPI processes via environment
* Second phase : Port all C unit tests to run with any number of MPI processes and allow users to run the entire test suite by specifying the number of MPI processes (This will be focused on a future issue, since most of the tests here are unit tests).

### PR DESCRIPTION
**Pull request by jayeshkrishna**
This PR includes changes to simplify PIO test creation and
enhancements to scale it to a user-specified number of MPI
processes.

Fixes #181

### CODE REVIEW

**Comment by jayeshkrishna** [comment]
Note that this should be considered a first stab at scaling the test suite. With this change the number of tests in the default test suite has increased from 90+ tests to 160+ tests. However we now test more combinations of PIO I/O processes and stride than before, which should be useful to us.

Moreover users can now run the suite on a larger cluster (e.g. To run the tests upto 128 procs, build PIO2 after setting PIO_TF_MAXNUMPROCS=128 - the test suite now generates multiple tests, for each test case, scaling the number of MPI processes by power of 2 until PIO_TF_MAXNUMPROCS)

**Comment by jayeshkrishna** [comment]
Shortcomings (e.g. scaling C tests, specify PIO root process etc) in this PR can be slated for a future issue/PR

File: tests/general/CMakeLists.txt [hunk 1]
```diff
@@ -64,8 +64,73 @@ endif()
 #  DEFINE THE TARGETS AND TESTS
 #==============================================================================
 
-# Test Timeout (4 min = 240 sec)
+# Test Timeout (8 min = 480 se)
```
Comment (dqwu) [inline tests/general/CMakeLists.txt]: typo: "480 se" ==> "480 sec"

File: tests/general/util/pio_tutil.F90 [hunk 2]
```diff
@@ -149,6 +149,9 @@ SUBROUTINE  PIO_TF_Init_(rearr)
     IF (pio_tf_num_io_tasks_ <= 1 .AND. pio_tf_stride_ > 1) THEN
        pio_tf_stride_ = 1
     END IF
+    IF (pio_tf_stride_ * (pio_tf_num_io_tasks_ - 1) >= pio_tf_world_sz_) THEN
```
Comment (dqwu) [inline tests/general/util/pio_tutil.F90]: Maybe we need a comment here stating that "assume root = 0"

File: tests/general/CMakeLists.txt (hunk 1 above)
Comment (jayeshkrishna) [reply tests/general/CMakeLists.txt]: Ok, will fix it

File: tests/general/util/pio_tutil.F90 (hunk 2 above)
Comment (jayeshkrishna) [reply tests/general/util/pio_tutil.F90]: Right now the testing framework does not allow changing the PIO root, we can refactor this in a future PR that adds that option.

**Comment by jayeshkrishna** [comment]
The number of tests increased from 97 to 165 and the nightly test times increased from ~10m to ~15m.

**Pull request merged**